src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -1803,3 +1804,31 @@ def delete_scope(self, scope_id: UUIDLike) -> GlobusHTTPResponse:
                     :ref: auth/reference/#delete_scope
         """
         return self.delete(f"/v2/api/scopes/{scope_id}")
+
+    def get_consents(self, identity_id: UUIDLike) -> GetConsentsResponse:
+        """
+        Look up consents for a user. Requires the ``view_consents`` scope.
```

Comment:
The `view_consents` should only be required if the `all=true`  query param is passed.  For the purposes of the cli client trying to guess what scopes are needed, it shouldn't need to include that query param since it is only interested in its own consents.  In fact, I'd strongly encourage not asking the user to consent to the "view_consents" scope if we don't need it as there is no reason for a client in most cases to see other consents, unless they are implementing functionality to manipulate other consents (e.g. in the globus webapp if we moved the manage consents functionality into it).

Reply:
Just getting back to this PR now.

This is a very good call out, thanks for catching that! We really only want to build out the consent tree rooted at our client for GlobusApp purposes.

I'll add `all` as an optional kwarg to this function (defaulting to False) & update the documentation to reflect this distinction.

Reply:
Updated the interface & called out in the documentation that most clients should not be requesting False.

Lemme know if you're uncomfortable with this change/wording.

Reply:
LGTM